core/src/main/java/org/elasticsearch/common/io/stream/StreamInput.java

```diff
@@ -826,6 +826,19 @@ public DateTimeZone readOptionalTimeZone() throws IOException {
     }
 
     /**
+     * Reads a {@link NamedWriteable} from the current stream with the given name. It is assumed that the caller obtained the name
```

Comment:
I'd add something like `Prefer {@link StreamInput#readNamedWriteable(Class)} and {@link StreamOutput#writeNamedWriteable(NamedWriteable)} unless you have a compelling reason to use this method instead.`.